cumulus/client/cirrus-executor/src/tests.rs

```diff
@@ -226,20 +246,30 @@ async fn execution_proof_creation_and_verification_should_work() {
 		let target_trace_root: Hash = intermediate_roots[target_extrinsic_index].into();
 		assert_eq!(target_trace_root, post_delta_root);
 
+		// Test `apply_extrinsic` verification on executor.
 		let execution_result = subspace_fraud_proof::check_execution_proof(
 			&charlie.backend,
 			&*charlie.code_executor,
 			charlie.task_manager.spawn_handle(),
 			&BlockId::Hash(parent_header.hash()),
 			&execution_phase,
 			post_delta_root,
-			storage_proof,
+			storage_proof.clone(),
 		)
 		.expect("Check extrinsic execution proof");
-
 		let post_execution_root =
 			execution_phase.decode_execution_result::<Header>(execution_result).unwrap();
 		assert_eq!(post_execution_root, intermediate_roots[target_extrinsic_index + 1].into());
+
+		// Test `apply_extrinsic` verification on farmer.
+		let fraud_proof = FraudProof {
+			parent_hash: parent_hash_alice,
+			pre_state_root: intermediate_roots[target_extrinsic_index].into(),
+			post_state_root: intermediate_roots[target_extrinsic_index + 1].into(),
+			proof: storage_proof,
+			execution_phase,
+		};
```

Comment:
Farmer is a different app, better not to confuse reader here.

Also this got me thinking, where is the proof here? I see proof of correct execution instead. I guess it is not the final form of it since it lacks offender?

Reply:
Yes, it's not the final version. The offender will be added once we start to take care of the slashing function.